### PR TITLE
TE5-T375 chapter3(A) Complete

### PR DESCRIPTION
**Pull request by vishal2001prajapati**

### CODE REVIEW

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 1]
```diff
@@ -5,11 +5,27 @@ import androidx.appcompat.app.AppCompatActivity
...
+import android.widget.Toast
 
 class MainActivity : AppCompatActivity() {
+    lateinit var button: Button
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        button = findViewById(R.id.button_ok)
+        button.setOnClickListener {
+            Toast.makeText(this, "This is a OK Button", Toast.LENGTH_LONG).show()
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt]: Move static string to string file

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 2]
```diff
@@ -387,28 +403,41 @@ class MainActivity : AppCompatActivity() {
...
+    private fun showtoast() {
+        val inflater: LayoutInflater = layoutInflater
+        val view: View = inflater.inflate(R.layout.custom_toast, findViewById(R.id.custom))
+        val toast: Toast = Toast(this)
+        toast.setGravity(Gravity.BOTTOM or Gravity.FILL_HORIZONTAL, 0, 0)
+        toast.duration = Toast.LENGTH_LONG
+        toast.view = view
+        toast.show()
+    }
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt:416]: extra empty line

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 3]
```diff
@@ -387,28 +403,41 @@ class MainActivity : AppCompatActivity() {
...
+
+    //sealed class ->third party can't access
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt:417]: code formtting

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 4]
```diff
@@ -387,28 +403,41 @@ class MainActivity : AppCompatActivity() {
...
-    override fun sub() {
-        Log.d("interface", "this is sub")
+        override fun sub() {
+            Log.d("interface", "this is sub")
+
+        }
     }
-}
\ No newline at end of file
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt:440]: code formatting

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -7,12 +7,161 @@
     tools:context=".MainActivity">
 
     <TextView
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: use appcompcat textview

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -7,12 +7,161 @@
...
+        android:textStyle="bold"
         app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintHorizontal_bias="0.442"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.04" />
 
+    <EditText
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: use AppCompact/androidx widgets in whole PR

File: app/src/main/res/layout/custom_toast.xml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="horizontal"
+    android:id="@+id/linear_layout"
+    android:background="@color/background_color"
+    android:padding="10dp"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
```
Comment (purvesh-d) [inline app/src/main/res/layout/custom_toast.xml]: code formtting

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/myapplication3/MainActivity.kt (hunk 1 above)
Comment (vishal2001prajapati) [reply app/src/main/java/com/example/myapplication3/MainActivity.kt]: done

File: app/src/main/java/com/example/myapplication3/MainActivity.kt (hunk 2 above)
Comment (vishal2001prajapati) [reply app/src/main/java/com/example/myapplication3/MainActivity.kt:416]: Done

File: app/src/main/java/com/example/myapplication3/MainActivity.kt (hunk 3 above)
Comment (vishal2001prajapati) [reply app/src/main/java/com/example/myapplication3/MainActivity.kt:417]: Done.

File: app/src/main/java/com/example/myapplication3/MainActivity.kt (hunk 4 above)
Comment (vishal2001prajapati) [reply app/src/main/java/com/example/myapplication3/MainActivity.kt:440]: Done.

File: app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (vishal2001prajapati) [reply app/src/main/res/layout/activity_main.xml]: Done.

File: app/src/main/res/layout/custom_toast.xml (hunk 7 above)
Comment (vishal2001prajapati) [reply app/src/main/res/layout/custom_toast.xml]: Done.

File: app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -6,13 +6,162 @@
...
+        android:layout_marginStart="24dp"
+        android:layout_marginEnd="24dp"
+        android:layout_marginBottom="160dp"
+        android:background="@drawable/ripple_effect"
+        android:paddingStart="15dp"
+        android:paddingEnd="20dp"
+        android:text="@string/ok"
+        android:textStyle="bold"
+        app:layout_constraintBottom_toBottomOf="parent"
+
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: extra empty lines

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 9]
```diff
@@ -381,34 +397,45 @@ class MainActivity : AppCompatActivity() {
...
-        obj5.add()
+        var calcutore = Calcutore()
+        calcutore.add()
+    }
+
+    private fun showtoast() {
+        val inflater: LayoutInflater = layoutInflater
+        val view: View = inflater.inflate(R.layout.custom_toast, findViewById(R.id.custom))
+        val toast: Toast = Toast(this)
+        toast.setGravity(Gravity.BOTTOM or Gravity.FILL_HORIZONTAL, 0, 0)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt]: static 0

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 10]
```diff
@@ -381,34 +397,45 @@ class MainActivity : AppCompatActivity() {
...
-    override fun add() {
-        var x: Int = 10
-        var y: Int = 12
-        var c: Int = 0
-        c = x + y
-        val d = Log.d("interface", c.toString())
-    }
+    class Calcutore : add, Sub {
+        override fun add() {
+            var x: Int = 10
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt]: static integer not allowed

File: app/src/main/java/com/example/myapplication3/MainActivity.kt [hunk 11]
```diff
@@ -381,34 +397,45 @@ class MainActivity : AppCompatActivity() {
...
+            var c: Int = 0
+            c = x + y
+            val d = Log.d("interface", c.toString())
+        }
 
-    override fun sub() {
-        Log.d("interface", "this is sub")
+        override fun sub() {
+            Log.d("interface", "this is sub")
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/MainActivity.kt]: extra empty line

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_main.xml [hunk 12]
```diff
@@ -6,13 +6,161 @@
...
+        android:paddingTop="20dp"
+        android:text="@string/hindi"
+        android:textColor="@color/text_color"
+        android:textSize="20sp"
+        android:textStyle="bold"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/chech_eng" />
+
+    <androidx.appcompat.widget.AppCompatTextView
+        android:id="@+id/gender"
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: proper id name, `text_gender ` or `txt_gender`

File: app/src/main/res/layout/activity_main.xml [hunk 13]
```diff
@@ -6,13 +6,161 @@
...
+    <RadioGroup
+        android:id="@+id/radio_gr"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        app:layout_constraintTop_toBottomOf="@+id/gender"
+        tools:ignore="MissingConstraints">
+
+        <androidx.appcompat.widget.AppCompatRadioButton
+            android:id="@+id/male"
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: id should be proper , check in whole file

File: app/src/main/res/layout/custom_toast.xml [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+    android:layout_height="match_parent"
+    android:background="@color/background_color"
+    android:orientation="horizontal"
+    android:padding="10dp">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="20dp"
+        android:text="This is a Custom Toast"
```
Comment (purvesh-d) [inline app/src/main/res/layout/custom_toast.xml]: static string

**Review by purvesh-d — CHANGES_REQUESTED** [review]
- Check id of all the widgets in all the files and give proper name.

**Pull request merged**